native/src/tablecloth.ml

```diff
@@ -6,6 +6,133 @@ let ( << ) (f1 : 'b -> 'c) (f2 : 'a -> 'b) : 'a -> 'c = fun x -> x |> f2 |> f1
 
 let identity (value : 'a) : 'a = value
 
+module Array = struct
+  let empty : 'a array = [||]
+
+  let singleton (a : 'a) : 'a array = [|a|]
+
+  let length (a : 'a array) : int = Base.Array.length a
+
+  let isEmpty (a : 'a array) : bool = length a = 0
+
+  let is_empty = isEmpty
+
+  let initialize ~(length : int) ~(f : int -> 'a) = 
+    if length <= 0 then empty else Base.Array.init length ~f
+  
+  let repeat ~(length : int) (e : 'a) : 'a array = 
+    if length <= 0 then empty else Base.Array.init length ~f:(fun _ -> e)
+
+  let range ?(from = 0) (to_ : int) : int array =
+    Base.Array.init (max 0 (to_ - from)) ~f:(fun i -> i + from)
+
+  let fromList (l: 'a list) : 'a array = Base.List.to_array l
+
+  let from_list = fromList
+
+  let toList (a: 'a array) : 'a list = Base.Array.to_list a
+
+  let to_list = toList
+
+  let toIndexedList xs = 
+    Base.Array.fold_right xs ~init:(length xs - 1, []) ~f:(fun x (i, acc) -> 
+      (i - 1, ((i, x) :: acc)))
+    |> Base.snd
+  
+  let to_indexed_list = toIndexedList
+
+  let get ~index a = 
+    if index >= 0 && index < length a then Some (Base.Array.get a index) else None
+
+  let set ~index ~value a = Base.Array.set a index value
+
+  let sum (a : int array) : int = Base.Array.fold a ~init:0 ~f:( + )
+
+  let floatSum (a : float array) : float = Base.Array.fold a ~init:0.0 ~f:( +. )
+
+  let float_sum = floatSum
+
+  let filter ~(f : 'a -> bool) (a : 'a array) : 'a array = Base.Array.filter a ~f
+
+  let map ~(f : 'a -> 'b) (a : 'a array) : 'b array = Base.Array.map a ~f
+
+  let mapWithIndex  ~(f : 'int -> 'a -> 'b) (a : 'a array) : 'b array = Base.Array.mapi a ~f
+  
+  let map_with_index = mapWithIndex
+  
+  let mapi = mapWithIndex 
+
+  let map2 ~(f : 'a -> 'b -> 'c) (a : 'a array) (b : 'b array) : 'c array =
+    let minLength = min (length a) (length b) in
+    Base.Array.init minLength ~f:(fun i -> f a.(i) b.(i))
+
+  let map3 ~(f : 'a -> 'b -> 'c -> 'd) (arrayA : 'a array) (arrayB : 'b array) (arrayC : 'c array) : 'd array =
+    let minLength : int = Base.Array.fold ~f:Base.min ~init:(length arrayA) [|length arrayB; length arrayC|] in
+    Base.Array.init minLength ~f:(fun i -> f arrayA.(i) arrayB.(i) arrayC.(i))
+
+  let flatMap ~f a = Base.Array.concat_map a ~f
+
+  let flat_map = flatMap
+
+  let find ~(f : 'a -> bool) (a : 'a array) : 'a option = Base.Array.find a ~f
+
+  let append (a : 'a array) (a' : 'a array)  : 'a array = Base.Array.append a a'
+
+  let concatenate  (al : 'a array array) : 'a array = Base.Array.concat (Base.Array.to_list al)
+
+  let flatten = concatenate
+
```

Comment:
you can remove flatten

Reply:
But it is listed in the `mli`? Or did you mean its an alias of concatenate?

Reply:
https://github.com/darklang/tablecloth/issues/22